compiler/modules/front.syntax/tests/posTree.bal

```diff
@@ -0,0 +1,585 @@
+type SyntaxNode TerminalSyntaxNode|NonTerminalSyntaxNode|IdentifierSyntaxNode|FixedSyntaxNode;
+
+// pr-todo: may be we need a union of Node that has a ASTNode to simplify stuff
+type TerminalSyntaxNode record {|
+    AstNode astNode;
+    string? token;  // if non-nil expect the token to be this
+|};
+
+type NonTerminalSyntaxNode record {|
+    AstNode astNode;
+    SyntaxNode[] childNodes;
+|};
+
+type RootSyntaxNode record {|
+    ModulePart part;
```

Comment:
Would it be simpler to put a startPos and endPos in ModulePart?

Reply:
fixed